youtube_dl/extractor/cda.py

```diff
@@ -0,0 +1,69 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+import re
+
+from .common import InfoExtractor
+from ..utils import (
+    decode_packed_codes,
+    ExtractorError,
+    parse_duration,
+)
+
+
+class CDAIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:(www|ebd)\.)?cda\.pl/(video|[0-9]+x[0-9]+)/(?P<id>[0-9a-z]+)'
+    _TEST = {
+        'url': 'http://www.cda.pl/video/5749950c',
+        'md5': '6f844bf51b15f31fae165365707ae970',
+        'info_dict': {
+            'id': '5749950c',
+            'ext': 'mp4',
+            'title': 'Oto dlaczego przed zakrętem należy zwolnić.',
+            'duration': 39,
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage("http://ebd.cda.pl/0x0/" + video_id, video_id)
+
+        if 'Ten film jest dostępny dla użytkowników premium' in webpage:
+            raise ExtractorError('This video is only available for premium users.', expected=True)
+
+        title = self._html_search_regex(r'<title>(.+?)<\/title>', webpage, 'title', fatal=False)
+        durations = []
+
+        def _get_format(page):
+            unpacked = decode_packed_codes(page)
+            duration = self._search_regex(r"duration:\\'(.+?)\\'", unpacked, 'duration', default=None)
+            if duration:
+                durations.append(parse_duration(duration))
+
+            m = re.search(r'<a data-quality="(?P<format_id>[^"]+)" href="[^"]+" class="quality-btn quality-btn-active">(?P<height>[0-9]+p)<\/a>',
+                          page, re.DOTALL)
+
+            return {
+                'url': self._search_regex(r"url:\\'(.+?)\\'", unpacked, 'url'),
+                'format_id': m.group('format_id'),
+                'height': m.group('height').rstrip('p')
+            }
+
+        formats = [_get_format(webpage)]
+
+        pattern = re.compile(r'<a data-quality="[^"]+" href="([^"]+)" class="quality-btn">([0-9]+p)<\/a>')
+        for version in re.findall(pattern, webpage):
+            webpage = self._download_webpage(version[0], video_id, 'Fetching %s version information' % version[1])
+            formats.append(_get_format(webpage))
+
+        self._sort_formats(formats)
+
+        if (len(durations) <= 0) or (durations.count(durations[0]) != len(durations)):
+            durations.insert(0, 0)
```

Comment:
My idea was to validate if returned duration is the same for each format, and set it to zero otherwise (or when there is no duration). But it is indeed not needed. Duration is not critical. I've changed it not so that it will remember first valid duration it find or set to `None` if not found  as you suggested.